### ISSUE TITLE
Passive Mode Agent

### ISSUE THREAD

**Issue by fredo** [beamer-bridge/beamer#2012]
Cerated out of #1757 

Recent events have shown that the agent needs to be able to run in a safe mode where it maintains the current position but does not engage in new positions.
The main reason for it is, that the agent is required to have a certain amount of ETH to function properly. But currently the agent is designed to only spend ETH, eventually depleting its ETH funds.

With the minimum source balance, an LP could already mitigate entering new positions but I consider this more of a hack. 

## Passive Mode
I see the spec of the passive mode as follows: 

* Don't fill any new requests
* Continue to run the agent
* Claim/Challenge/withdraw should be executed like in the normal mode
* Agent should not enter new challenges
* Agent should log if their are no outstanding transactions anymore and exit
* If Agent starts in passive mode and there are no outstanding transactions sam as the above
* Flag in the config to enable passive mode

**Comment by istankovic** [comment]
> Agent should not enter new challenges

This is just for other agents' claims, right?

I think it would be helpful to be more specific here. I can think of at least these situations (feel free to add more!):

- [ ] a new challenge arrives for a claim made by the agent
- [ ] a new challenge arrives for a claim not made by the agent
- [ ] next turn of a challenge game in which the agent participates
- [ ] next turn of a challenge game in which the agent does not participate

Please check those cases where you think a passive-mode agent should take action.

**Comment by istankovic** [comment]
> Should ignore the whitelist check

What does this mean?

**Comment by fredo** [comment]
> next turn of a challenge game in which the agent does not participate

Nothing happens here even in the normal mode

**Comment by fredo** [comment]
I don't think that it makes a lot of sense to list all possible situations, rather I'd like to extend on the invariants.
The goal of this mode is to successfully exit the system. In a bit more detail this means that the agent has no stake/tokens whatsoever in the contracts anymore. In order to achieve this state, the agent should not enter new positions.

* There should be a deterministic end of this mode. This means the agent should not engage in new positions which make it possible to run forever (continuous overlapping new positions like new challenges or new fills). Specifically fills or challenges are new entry points for further interaction with the system.
* The agent should follow the protocol on all claims to receive his funds back eventually.
* The agent will input new funds into the system if and only if it leads to finalizing outstanding positions (claims) according to the protocol to receive his funds back(Here the only case is if the agent gets challenged on one of his existing claims aka where he is already participating)
* As soon as the agent has no funds left in all Beamer contracts, the software will exit


If you are uncertain with a specific case, please look at the above statements and see if they answer your case. If not we should extend the list of invariants.
